google-cloud-logging/src/main/java/com/google/cloud/logging/MonitoredResourceUtil.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.logging;
+
+import com.google.cloud.MetadataConfig;
+import com.google.cloud.MonitoredResource;
+import com.google.cloud.ServiceOptions;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Monitored resource construction utilities to detect resource type and add labels.
+ */
+abstract class MonitoredResourceUtil {
```

Comment:
i had done this to avoid instantiation. added private constructor instead. i've also marked this public. logging framework adapter implementations like slf4j (that will not be part of google-cloud-java) will need access to the resource construction : `getResource` and `getResourceEhancers` methods.